frontend/src/js/components/viewer/Viewer.tsx

```diff
@@ -195,6 +215,7 @@ class Viewer extends React.Component<Props, State> {
     componentWillUnmount() {
         document.title = "Giant";
         this.props.resetResource();
+        this.props.resetPages();
```

Comment:
Without this the old page state hangs around and we can't properly calculate the fallback logic if you click back to Search and then into the Viewer again via a seach result